### PR TITLE
Make converter optional

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR makes the converter optional in OTel agent. It is enabled by default, but users now have an option to disable it in agent config.

### Describe how to test/QA your changes

Run the agent with the `otelcollector.converter.enabled` set to false (or via env `DD_OTELCOLLECTOR_CONVERTER_ENABLED=false` and ensure the collector config does not get enhanced)

### CODE REVIEW

File: comp/otelcol/collector/impl/collector.go [hunk 1]
```diff
@@ -127,7 +129,9 @@ func NewComponent(reqs Requires) (Provides, error) {
 	}
 	addFactories(reqs, factories)
 
-	err = reqs.ConfigStore.AddConfigs(newConfigProviderSettings(reqs, false), newConfigProviderSettings(reqs, true), factories)
+	converterEnabled := reqs.Config.Get("otelcollector.converter.enabled").(bool)
```
Comment (dineshg13) [inline comp/otelcol/collector/impl/collector.go:132]: use `GetBool`

**Review by hush-hush — APPROVED** [review]
Looks good for the files owned by ASC

**Review by ankitpatel96 — APPROVED** [review]

**Pull request merged**